spec/jobs/update_diaper_partner_job_spec.rb

```diff
@@ -53,4 +53,13 @@
       end
     end
   end
+
+  describe ".partner_attributes" do
+    it "includes the organization_email in the output" do
+      @partner = create(:partner)
+      organization_email = @partner.organization.email
+      update_partner_job = UpdateDiaperPartnerJob.new
+      expect(update_partner_job.partner_attributes(@partner)).to have_value(organization_email)
+    end
+  end
```

Comment:
Here is a spec that does a few things slightly differently that I think can be beneficial.
- We are adding test coverage to ensure that the proposed method (or equivalent logic) is being used in the use case. In this case, that we sure `DiaperPartnerClient` is sending the right stuff.
- Remove dependency on the database entirely and active records. Some schools of thought consider the DB to be an implementation detail that can be replaced in testing.
- Added a little extra test to ensure `invited!` is called on the partner.

```
  describe '#perform(partner_id)' do
    subject { described_class.new.perform(partner_id) }
    let(:partner_id) { Faker::Number.number }

    context 'when the email feature is active' do
      # I'am only going handle the branch of logic when it email is active
      let(:fake_partner) { instance_double(Partner, attributes: { fake: 'data' }) }

      before do
        # Force email to be enabled
        allow(Flipper).to receive(:enabled?).with(:email_active).and_return(true)

        allow(Partner).to receive(:find).with(partner_id).and_return(fake_partner)

        # Pre-program all interactions with the fake partner. This is to
        # remove any dependencies on what Partner actually does.
        fake_invitation_text = Faker::Lorem.sentences
        fake_organization_email = Faker::Internet.email

        allow(fake_partner).to receive_message_chain(:organization, :invitation_text).and_return(fake_invitation_text)
        allow(fake_partner).to receive_message_chain(:organization, :email).and_return(fake_organization_email)

        # The below will fail the test if `.with(...)` doesn't
        # match what is actually happening.
        allow(DiaperPartnerClient).to receive(:post)
          .with(fake_partner.attributes.merge({ organization_email: fake_organization_email }), fake_invitation_text)
          .and_return({})

        allow(fake_partner).to receive(:invited!)
      end

      it 'should have called invited! on the partner' do
        subject
        expect(fake_partner).to have_received(:invited!)
      end
    end
  end
```

I realize this is a pretty big change request. However, I think this might just work out of the box.

Reply:
Yeah this seems good, thanks!